packages/cli/src/commands/did/from-seed.ts

```diff
@@ -21,33 +16,28 @@ export default class DIDFromSeed extends Command<Flags, { didKeySeed: string }>
     },
   ]
 
-  static flags = {
-    ...Command.flags,
-    'did-key-seed': Flags.string({
-      description: 'a random 32-bit seed represented as a base16 string',
-      required: false,
-    }),
-  }
-
   async run(): Promise<void> {
     this.spinner.start('Creating DID...')
-    if (this.args.didKeySeed === undefined && this.flags['did-key-seed'] === undefined) {
+    const possibleSeedInputs = [this.stdin, this.args.didKeySeed, this.flags['did-key-seed']]
+    if (possibleSeedInputs.every((input) => input === undefined)) {
       this.spinner.fail(
-        `You need to pass the seed parameter either as a positional arg or as a flag value`
+        `You need to pass the seed parameter as a positional arg, as a flag value, via stdin or as the DID_KEY_SEED environmental variable`
       )
       return
-    } else if (this.args.didKeySeed !== undefined && this.flags['did-key-seed'] !== undefined) {
+    } else if (possibleSeedInputs.filter((input) => input !== undefined).length > 1) {
       this.spinner.fail(
-        `Don't pass the seed parameter as both a positional arg and as a flag value`
+        `Don't pass the seed parameter in more than one way out of: arg, flag, stdin, DID_KEY_SEED environmental variable`
       )
       return
     }
     try {
-      const hexString = this.args.didKeySeed || (this.flags['did-key-seed'] as string) || ''
+      const hexString = possibleSeedInputs.find((input) => input !== undefined) || ''
       const seed = fromString(hexString, 'base16')
       const did = new DID({ provider: new Ed25519Provider(seed), resolver: this.resolverRegistry })
       await did.authenticate()
-      this.spinner.succeed(`Created DID ${chalk.cyan(did.id)}`)
+      this.spinner.succeed(`Creating DID... Done!`)
+      // Logging the DID to stdout, so that it can be piped using standard I/O or redirected to a file
+      this.log(chalk.cyan(did.id))
```

Comment:
Does using `chalk` change the output when piped? I think we can remove it if there's any risk that would pipe extra formatting characters.

Reply:
It didn't spoil anything in my manual tests, but I'll remove it, as the whole thing needs more testing in general.